From: Agathe Porte <agathe.porte@oss.qualcomm.com>
Date: Mon, 10 Aug 2026 16:28:21 +0200
Subject: move check-camx-overlay.sh to libexec

This script should not be present in /usr/bin/ for such a limited
use-case, and should be moved into libexec instead.

Use a subdirectory named after the binary package to avoid polluting the
libexec root given this script has no prefix.

Forwarded: https://github.com/qualcomm/camera-service/pull/25
Signed-off-by: Agathe Porte <agathe.porte@oss.qualcomm.com>
---
 qmmf-server/CMakeLists.txt                     | 2 +-
 qmmf-server/service/cam-server-base.service.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/qmmf-server/CMakeLists.txt b/qmmf-server/CMakeLists.txt
index 6760343..27954c8 100644
--- a/qmmf-server/CMakeLists.txt
+++ b/qmmf-server/CMakeLists.txt
@@ -42,7 +42,7 @@ install(FILES
 
 install(PROGRAMS
         ${CMAKE_CURRENT_SOURCE_DIR}/check-camx-overlay.sh
-        DESTINATION bin)
+        DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/qmmf-server)
 
 if(HAVE_ANDROID_UTILS)
 target_link_libraries(cam-server log cutils utils binder)
diff --git a/qmmf-server/service/cam-server-base.service.in b/qmmf-server/service/cam-server-base.service.in
index 8c7aa4d..e9d941d 100644
--- a/qmmf-server/service/cam-server-base.service.in
+++ b/qmmf-server/service/cam-server-base.service.in
@@ -43,7 +43,7 @@ RuntimeDirectory=cam_server
 RuntimeDirectoryMode=0775
 RuntimeDirectoryPreserve=yes
 CacheDirectory=camera
-ExecCondition=/usr/bin/check-camx-overlay.sh
+ExecCondition=/usr/libexec/qmmf-server/check-camx-overlay.sh
 ExecStart=/usr/bin/cam-server
 
 [Install]
